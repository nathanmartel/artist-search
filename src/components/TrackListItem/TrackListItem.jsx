import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const TrackListItem = ({ id, title, length }) => {
  
  return (
    <li key={id}>
      <Link to={`/song/${id}`}>
        {title} ({length})
      </Link>
    </li>
  );
};

TrackListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
};

export default TrackListItem;
