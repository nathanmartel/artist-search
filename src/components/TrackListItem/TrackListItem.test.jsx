import React from 'react';
import { shallow } from 'enzyme';
import TrackListItem from './TrackListItem';

describe('TrackListItem component', () => {
  it('renders TrackListItem', () => {
    const wrapper = shallow(<TrackListItem id={'idstring'} title={'Fitter Happier'} length={'12345'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
