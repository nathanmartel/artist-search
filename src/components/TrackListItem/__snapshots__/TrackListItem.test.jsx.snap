// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListItem component renders TrackListItem 1`] = `
<li
  key="idstring"
>
  <Link
    to="/song/idstring"
  >
    Fitter Happier
     (
    12345
    )
  </Link>
</li>
`;
